docs/docs-components/data/types.js

```diff
@@ -12,3 +12,64 @@ export type DesignOverview = {|
   path: string,
   svg: SVGElement,
 |};
+
+type StatusType = 'notAvailable' | 'partial' | 'planned' | 'ready';
+
+/**
+ * All components should be in a single list. That list can be filtered by category/platform/etc where needed.
+ */
+
+type ComponentAccessibility = {|
+  a11yComprehension?: StatusType,
+  a11yNavigation?: StatusType,
+  a11yScreenreader?: StatusType,
+  a11yVisual?: StatusType,
+  summary?: StatusType,
+|};
+
+type ComponentCategory =
+  | 'Actions'
+  | 'Avatars'
+  | 'Building blocks'
+  | 'Controls'
+  | 'Data'
+  | 'Fields and forms'
+  | 'Help and guidance'
+  | 'Indicators'
+  | 'Loading'
+  | 'Messaging'
+  | 'Navigation'
+  | 'Overlays'
+  | 'Pins and imagery'
+  | 'Structure'
+  | 'Text'
+  | 'Utilities';
+
+type PlatformObjType<T> = {|
```

Comment:
The new data structure splits out most fields by platform (and we're treating Figma as another platform to allow for Figma-only components). This gives us a lot of flexibility for different values by platform.

Given that most fields _aren't_ platform-specific, there are a couple of approaches we can take:
- Since all platform fields are nullable, default to `web` values if others aren't present
- Duplicate values manually to each platform
I think the first approach is much, much better — it uses a single source of truth and prevents a lot of tedious copy/pasting (and bugs from inevitable drift).

This also cuts down on a lot of boilerplate inherent in the current data structure. If a component doesn't exist on Android, we don't need to add an object full of null values — we just don't add `android` at all in the `status` object.